File: task3.py
from flask import Flask, request, render_template, redirect, url_for
import sqlite3

app = Flask(__name__)

# Define the path to the SQLite database
DATABASE = 'database.db'

def connect_db():
    return sqlite3.connect(DATABASE)

@app.route('/')
def index():
    return render_template('task3_index.html') # manually changed the template

@app.route('/check_name', methods=['POST'])
def check_name():
    name = request.form['name']
    db = connect_db()
    cursor = db.cursor()

    # Check if the name exists in the database
    cursor.execute("SELECT COUNT(*) FROM names WHERE name = ?", (name,))
    count = cursor.fetchone()[0]

    db.close()

    return render_template('task3_result.html', name=name, count=count) # manually changed the template

if __name__ == '__main__':
    app.run(debug=True)

File: task3.php
<?php
$host = 'localhost'; // Change this to your database host if it's different
$username = 'your_username'; // Replace with your database username
$password = 'your_password'; // Replace with your database password
$database = 'name_checker';

// Create a database connection
$conn = new mysqli($host, $username, $password, $database);

// Check the connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

// Function to check if a name exists in the database
function checkName($name, $conn) {
    $name = mysqli_real_escape_string($conn, $name);
    $query = "SELECT * FROM names WHERE name='$name'";
    $result = $conn->query($query);

    if ($result->num_rows > 0) {
        return true; // Name found in the database
    } else {
        return false; // Name not found in the database
    }
}

if (isset($_POST['submit'])) {
    $nameToCheck = $_POST['name'];
    $nameExists = checkName($nameToCheck, $conn);

    if ($nameExists) {
        echo "Name '$nameToCheck' is present in the database.";
    } else {
        echo "Name '$nameToCheck' is not found in the database.";
    }
}

$conn->close();
?>

<!DOCTYPE html>
<html>
<head>
    <title>Name Checker</title>
</head>
<body>
    <h1>Check if a Name is in the Database</h1>
    <form method="POST" action="">
        <label for="name">Enter a name: </label>
        <input type="text" name="name" required>
        <input type="submit" name="submit" value="Check">
    </form>
</body>
</html>
